src/main/java/com/sun/librarymanagement/domain/repository/UserRepository.java

```diff
@@ -0,0 +1,26 @@
+package com.sun.librarymanagement.domain.repository;
+
+import com.sun.librarymanagement.domain.entity.UserEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface UserRepository extends JpaRepository<UserEntity, Long> {
+
+    Optional<UserEntity> findByEmail(String email);
+
+    Optional<UserEntity> findByUsername(String username);
+
+    Optional<UserEntity> findByVerifyToken(String token);
+
+    @Query("SELECT u FROM UserEntity u WHERE u.username = :username OR u.email = :email")
+    List<UserEntity> findByUsernameOrEmail(
+        @Param("username") String username,
+        @Param("email") String email
+    );
```

Comment:
```suggestion
List<UserEntity> findByUsernameOrEmail(String username, String email);
```
thanh niên nghĩ gì về line code này